### ISSUE TITLE
Liquid Exception: No header received back - Mac OS X

### ISSUE THREAD

**Issue by jhulst** [jekyll/jekyll#2604]
Trying to get started in Jekyll and ran

`jekyll new test`
and then
`jekyll serve`

as directed in the startup documentation results in

```
Configuration file: /Users/jhulst/test/_config.yml
            Source: /Users/jhulst/test
       Destination: /Users/jhulst/test/_site
      Generating...
  Liquid Exception: No header received back. in _posts/2014-07-15-welcome-to-jekyll.markdown
jekyll 2.1.1 | Error:  No header received back.
```

Ran a Gem system update and also updated all packages, same result. I'm just getting started in Jekyll, so I'm sure I'm doing something wrong.

**Comment by Rafeh01** [comment]
Thank you for your comments @jhulst. That helped. I simply created an environment in my anaconda for python 2.7 and used that to host on jekyll using ``jekyll serve``.

**Comment by gerrywastaken** [comment]
@parkr Shouldn't these issues be handled in a more graceful way? Such as catching the "Failed to get header" exception and then outputting a message that tells user that Python 3 isn't supported and that they should switch to rogue?

This issue had so many comments that the thread was locked:
https://github.com/jekyll/jekyll/issues/1181

This is only going to get worse as more people migrate to Python 3

**Comment by DirtyF** [comment]
@gerrywastaken  or upgrade to Ubuntu 16.04 who comes with Python 3 installed by default.

**Comment by parkr** [comment]
> @parkr Shouldn't these issues be handled in a more graceful way? Such as catching the "Failed to get header" exception and then outputting a message that tells user that Python 3 isn't supported and that they should switch to rouge?

@gerrywastaken If you'll submit a PR, I'll review it!

**Comment by envygeeks** [comment]
@DirtyF that's wrong.  Ubuntu 16.04 comes with Python2 and Python3.

**Comment by envygeeks** [comment]
Let me rephrase that, Ubuntu standard comes with Python2 and Python3 by default.  And Ubuntu server *still* comes with Python2, as our fresh installs of 16.04 still have Python2 on the manifest.  I know they slated it for being Python3 only but apparently that didn't work out.

```
docker@gcec281:~# dpkg --get-selections | grep python2
libpython2.7:amd64				install
libpython2.7-minimal:amd64			install
libpython2.7-stdlib:amd64			install
python2.7					install
python2.7-minimal				install
```

That server was recently redone with fresh Ubuntu since we ops that way.

**Comment by gerrywastaken** [comment]
@parkr I'll be honest. This issue is not important to me and so it wouldn't be worth the time given I don't know your code base. I thought it would be important to the people running this project considering all the users running into this issue:

https://github.com/jekyll/jekyll/issues/1181 <- LOCKED Jul 4, 2015 because too many people said they had the issue
https://github.com/jekyll/jekyll/issues/925#issuecomment-21343682
https://github.com/jekyll/jekyll/issues/1379#issuecomment-22116142
https://github.com/jekyll/jekyll/issues/3873
https://github.com/jekyll/jekyll/issues/2017

Anyway, no worries.

**Comment by DirtyF** [comment]
@envygeeks Thanks for the clarification, I was mistaken by [Ubuntu Release Page](https://wiki.ubuntu.com/XenialXerus/ReleaseNotes#Python_3), I trust your experience on that.

**Comment by envygeeks** [comment]
@DirtyF you just reminded me of why I got beef with Ubuntu 16.04 >.> I'm on Skylake and 4.4 had some pretty serious bugs on Skylake so I have to backport 4.5 myself.
